### ISSUE TITLE
content in contentLoaded is always undefined

### ISSUE THREAD

**Issue by ben-qnimble** [facebook/docusaurus#10574]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

I want to use the [contentLoaded](https://docusaurus.io/docs/next/api/plugin-methods/lifecycle-apis#contentLoaded]) api to get the content of my site. 

I created the following plugin:

```js
module.exports = function getContent(...pluginArgs) {
  return {
      name:"testplugin",
      async contentLoaded({ content, actions }) {
          console.log("contentLoaded's content: ",content);
      },
  };
}

```
and added it to the plugins configuration in docusaurus.config.js. When I run yarn build, I get in the console:

```
contentLoaded's content:  undefined
```

but I'd expect to get the content of the site. Am I not using the async contentLoaded function correctly? And how can I get access to the content of the site for use in a plugin?

### Reproducible demo

https://codesandbox.io/p/devbox/green-firefly-37w5c9

### Steps to reproduce

Create plugin such as

```js
module.exports = function getContent(...pluginArgs) {
  return {
      name:"testplugin",
      async contentLoaded({ content, actions }) {
          console.log("contentLoaded's content: ",content);
      },
  };
}
```

try to use content. 

### Expected behavior

I would expect content to show the data associated with the site so that it can be parsed, etc.

### Actual behavior

content is always undefined.

### Your environment

- Public source code:  https://codesandbox.io/p/devbox/green-firefly-37w5c9
- Public site URL:    https://codesandbox.io/p/devbox/green-firefly-37w5c9
- Docusaurus version used:  3.5.2  
- Environment name and version (Node.js 18):
- Operating system and version: 

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
Each plugin can only get its own content. If you didn't define `loadContent`, then you will receive no content. It's not possible for a plugin to get content from other plugins, other than possibly in `postBuild` you can inspect the HTML build output etc.

I'm closing this for now since it's not a bug. If you have questions about how to implement a particular use case, please use GH discussions.

**Comment by slorber** [comment]
Technically there's a secret allContentLoaded() hook you can use that received the whole site plugins content aggregated. It's used by our debug plugin. 

But it's internal and I'm likely to remove it,  but if I ever do there will be an alternative option such as accessing the site content from the upcoming Server Components. So for an app I think it's relatively safe to implement this lifecycle until we have a better option.

**Comment by ben-qnimble** [comment]
Yeah, I was using allContentLoaded for my plugin when that was one of the arguments to contentLoaded (maybe from docusaurus 2.x?) and it worked fine for a while. But then it stopped working when I upgraded from 3.1 to 3.5 and I saw what I was doing was not anywhere in the documentation, so clearly that was asking for trouble. I ended up seeing your reference for doing this ([discussion](https://github.com/facebook/docusaurus/issues/4138) [example](https://stackblitz.com/edit/github-crattk?file=custom-blog-plugin.js,docusaurus.config.js,src%2Fcomponents%2FHome%2Findex.js,.gitignore) ) and that was super helpful. Using that code as a template, I was able to recreate my plugin but in a supported / working way. Thanks!
